state/machine.go

```diff
@@ -1855,12 +1820,15 @@ func (m *Machine) SetStatus(statusInfo status.StatusInfo) error {
 	case status.Pending:
 		// If a machine is not yet provisioned, we allow its status
 		// to be set back to pending (when a retry is to occur).
-		_, err := m.InstanceId()
-		allowPending := errors.Is(err, errors.NotProvisioned)
-		if allowPending {
-			break
-		}
-		fallthrough
+
+		// TODO(nvinuesa): we need to add back the check for provisioned and
+		// if it's not then the machine goes right to status.DOWN:
+		// _, err := m.InstanceId()
+		// allowPending := errors.Is(err, errors.NotProvisioned)
+		// if allowPending {
+		// 	break
+		// }
+		// fallthrough
```

Comment:
Ensure we have a jira card associated with this.

Reply:
We do: https://warthogs.atlassian.net/browse/JUJU-6875